ChelseasFieldJournal/src/GUI/GuiDialogToolJournal.cs

```diff
@@ -0,0 +1,202 @@
+﻿using Cairo;
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Drawing;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Vintagestory.API.Client;
+using Vintagestory.API.Common;
+using Vintagestory.API.Common.Entities;
+using Vintagestory.API.Config;
+using Vintagestory.API.MathTools;
+using static Ele.ChelseasFieldJournal.ModConstants;
+
+namespace Ele.ChelseasFieldJournal
+{
+    public class GuiDialogToolJournal : GuiDialogGeneric
+    {
+        
+        private int leftPageNumber { get; set; }
+        private int rightPageNumber { get; set; }
+        private Item ItemBook { get; set; }
+        private int ChapterLeftPage { get; set; }
+
+        private int Chapter { get; set; }
+
+        private object currentLeftPage;
```

Comment:
I would make an interface `IPageEntry` and derive _all_ your page entries from it so you can avoid using `object` and casting, potentially causing an InvalidCastException.

Reply:
done